File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*; 

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        LinkedList<String> retire = new LinkedList<>(Arrays.asList(participant)); // 일반 배영을 ArrayList 로 변환. 
        retire.sort(Comparator.naturalOrder()); // comparator 오름차순.
        Arrays.sort(completion);
        for (int i = 0 ; i < completion.length ; i++) {
            if (retire.get(0).equals(completion[i])) {
                retire.remove(0);
            }
            else if (retire.get(1).equals(completion[i])) {
                retire.remove(1);
            }
        }
        answer = retire.get(0);
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
  const map = new Map();

  // 참가자 +1
  for (const p of participant) {
    map.set(p, (map.get(p) || 0) + 1);
  }

  // 완주자 -1
  for (const c of completion) {
    map.set(c, map.get(c) - 1);
  }

  // 값이 남아있는 사람 찾기
  for (const [key, value] of map.entries()) {
    if (value > 0) {
      return key;
    }
  }
    
}